File: springboot-shop/src/main/java/com/ys/springbootshop/entity/Member.java
package com.ys.springbootshop.entity;

import com.ys.springbootshop.constant.Role;
import com.ys.springbootshop.dto.MemberFormDto;
import lombok.*;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.bind.annotation.CrossOrigin;

import javax.persistence.*;

/**
 * @author : ysk
 */
@Entity
@Table(name = "member")
@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Member extends BaseEntity{

    @Id
    @Column(name = "member_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Column(unique = true)
    private String email;

    private String password;

    private String address;

    @Enumerated(EnumType.STRING)
    private Role role;

    public static Member createMember(MemberFormDto memberFormDto,
                                      PasswordEncoder passwordEncoder) {
        return Member.builder()
                .name(memberFormDto.getName())
                .email(memberFormDto.getEmail())
                .address(memberFormDto.getAddress())
                .password(passwordEncoder.encode(memberFormDto.getPassword()))
                .role(Role.ADMIN)
                .build();
    }
}

File: graphql/graphqlex/web-1/src/main/kotlin/com/yscorp/web1/domain/Member.kt
package com.yscorp.web1.domain

import jakarta.persistence.*

@Entity
@Table(name = "member")
class Member {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @Column(name = "first_name")
    var firstName: String? = null

    @Column(name = "last_name")
    var lastName: String? = null

    @Enumerated(EnumType.STRING)
    @Column(name = "type")
    var type: MemberType = MemberType.STUDENT

    @Column(name = "contact")
    var contact: String? = null

}
